sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -797,6 +797,11 @@ private void unpackAndValidateId(String keyId) {
             URL url = new URL(keyId);
             String[] tokens = url.getPath().split("/");
             String endpoint = url.getProtocol() + "://" + url.getHost();
+
+            if (url.getPort() != -1) {
+                endpoint += ":" + url.getPort();
```

Comment:
If supported by versions of Java we support, 
have you looked at https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#getAuthority()? It will return both host and port (if port is provided)

```java
String endpoint = url.getProtocol() + "://" + url.getAuthority();
```

Reply:
Great suggestion, thank you Maor!

Reply:
Now that I think about it, `getAuthority()` also includes username and password information if present on the URL, would we want to include any of that in the key identifier in case somebody actually includes it?

Reply:
That makes sense! Sorry, was afk